common/core/app_registry.c

```diff
@@ -0,0 +1,183 @@
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+static const cy_app_desc_t *descriptors[REGISTRY_MAX_APPS] = {0};
+static uint16_t app_desc_count = 0;
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Compares values of two cy_app_desc_t instances.
+ *
+ * @param a_desc References to first descriptor
+ * @param b_desc Reference to second descriptor
+ * @return Whether the descriptors match.
+ * @retval true If the provided descriptor values match.
+ * @retval false If the provided descriptor values do not match.
+ */
+bool compare_descriptors(const cy_app_desc_t *a_desc,
+                         const cy_app_desc_t *b_desc);
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+bool compare_descriptors(const cy_app_desc_t *a_desc,
+                         const cy_app_desc_t *b_desc) {
+  bool status = true;
+  status &= (a_desc->version == b_desc->version);
+  status &= (memcmp(a_desc->name, b_desc->name, sizeof(cy_app_desc_t)) == 0);
+
+  return status;
+}
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+bool registry_add_app(const cy_app_desc_t *app_desc) {
+  bool status = false;
+
+  if (NULL == app_desc) {
+    return status;
+  }
+
+  // ensure app-id collision safety among descriptor list
+  if (NULL != registry_get_app_desc(app_desc->id)) {
+    return status;
+  }
+
+  // ensure registry storage does not overflow
+  ASSERT(app_desc_count < REGISTRY_MAX_APPS);
+
+  descriptors[app_desc_count - 1] = app_desc;
+  app_desc_count++;
+  status = true;
+  return status;
+}
+
+bool registry_find_app_desc(const cy_app_desc_t *app_desc, bool strict_check) {
+  bool status = false;
+  const cy_app_desc_t *desc = registry_get_app_desc(app_desc->id);
+
+  if (NULL == desc) {
+    return status;
+  }
+
+  // descriptor with same app-id is found
+  status = true;
+
+  if (strict_check) {
+    status = compare_descriptors(app_desc, desc);
+  }
+
+  return status;
+}
```

Comment:
Better to keep this check always strict as we won't be supporting apps with the same id but different names or versions simultaneously.

Reply:
changed